File: 05_arrays/05_12_single_number/single_number.js
const singleNumber = (arr = []) => {
    let xor = 0;
    for (const num of arr) {
        xor ^= num;
    }
    return xor;
}

const arr = [4, 1, 2, 1, 2];

console.log("Elements:-");
console.log(arr.join(' '));

const result = singleNumber(arr);
console.log("The single element is:", result);

File: 05_arrays/05_12_single_number/SingleNumber.java
public class SingleNumber {
    public static void main(String[] args) {
        int[] arr = {4, 1, 2, 1, 2};

        System.out.println("Elements:-");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        int result = singleNumber(arr);
        System.out.println("The single element is: " + result);
    }
    
    private static int singleNumber(int arr[]) {
        int xor = 0;
        for (int i : arr) {
            xor ^= i;
        }
        return xor;
    }
}
